server/stream.go

```diff
@@ -3778,17 +3783,21 @@ func (mset *stream) startingSequenceForSources() {
 	}
 
 	var smv StoreMsg
-	for seq := state.LastSeq; seq >= state.FirstSeq; seq-- {
-		sm, err := mset.store.LoadMsg(seq, &smv)
-		if err != nil || sm == nil || len(sm.hdr) == 0 {
+	for seq := state.LastSeq; ; {
+		sm, err := mset.store.LoadPrevMsg(seq, &smv)
+		if err == ErrStoreEOF || err != nil {
```

Comment:
Same here.

Reply:
EOF is the normal exit condition, but I was being pedantic. Could move it to be just err != nil but that does not read well in terms of intent.

Reply:
What do you suggest we do? Or leave as is for now?

Reply:
I think it's fine if it is trying to be clear that it's mostly trying to trap EOFs, the function doesn't have any way to bubble up other errors anyway.